src/core/utils/Renderer.ts

```diff
@@ -128,6 +130,13 @@ export class Renderer {
         if (!context.currentVNode.children.length && !context.currentVNode.properties.atomic) {
             this._insertFiller(context.currentVNode, element);
         }
+
+        // If a LINE_BREAK has no next sibling, it must be rendered as two BRs
+        // in order for it to be visible.
+        const nextSibling = context.currentVNode.nextSibling();
+        if (!__isSecondBR && context.currentVNode.type === VNodeType.LINE_BREAK && !nextSibling) {
+            return this._renderElement(context, true);
+        }
```

Comment:
What about the BasicHTMLRenderingEngine of the BR node doing that instead ? Makes more sense to me.

Reply:
You mean like this?

```typescript
export const BasicHtmlRenderingEngine = {
    /**
     * Render the given node to HTML.
     *
     * @param node
     */
    render: function(node: VNode): HTMLElement[] {
        const tagName = getRenderingTagName(node.type);
        const element = document.createElement(tagName);
        if (node.type === VNodeType.LINE_BREAK && !node.nextSibling()) {
            return [element, element.cloneNode()];
        }
        return element;
    },
};
```
and modify `_renderElement` to handle lists of elements?

Reply:
I ended up doing that, and the `_insertFiller` thing. I put the part where I return a `DocumentFragment` in a separate commit so it can be used by the other two commits.